src/main/java/com/owncloud/android/lib/common/utils/Log_OC.java

```diff
@@ -35,72 +111,73 @@ public static String getLogPath() {
         return mNextcloudDataFolderLog;
     }
 
+    /**
+     * Plug your own logger implementation. Call this as early as possible.
+     *
+     * @param adapter Custom logger implementation.
+     */
+    public static void setLoggerImplementation(Adapter adapter) {
+        impl = adapter;
+    }
+
     public static void i(String TAG, String message) {
-        Log.i(TAG, message);
-        appendLog(TAG + ": " + message);
+        impl.i(TAG, message);
     }
 
     public static void i(Object object, String message) {
         i(object.getClass().getSimpleName(), message);
```

Comment:
I added this some time ago to not have to use TAG over and over again.
Do we want to have this also in the new interface?

Reply:
Yeah, I used to do that too - it simplifies on-desk debugging, but doesn't work well for field deployments.

My change is meant to be backward compatible and since this is a library and I cannot know who uses it, I didn't dare to change the interface.

Ideally, when diagnosing issues, we'd like ot have more sophisticated tagging by subsystem, such as "sync" or "mediaplayer", to narrow down logs to a particular domain, not only a single class. This requires a bit different approach.

I definitely plan to revisit this, but I'm not ready to comment.

Reply:
Ok, then we keep it this way.